src/mysql_k8s_helpers.py

```diff
@@ -267,78 +267,83 @@ def execute_backup_commands(
         s3_directory: str,
         s3_access_key: str,
         s3_secret_key: str,
+        s3_endpoint: str,
     ) -> Tuple[str, str]:
         """Executes the run_backup.sh script in the container with the given args."""
-        nproc_commands = "nproc".split()
+        nproc_command = "nproc".split()
 
-        make_temp_dir_commands = "mktemp --tmpdir --directory xtra_backup_XXXX".split()
+        make_temp_dir_command = "mktemp --tmpdir --directory xtra_backup_XXXX".split()
 
         try:
-            process = self.container.exec(nproc_commands)
+            process = self.container.exec(nproc_command)
             nproc, _ = process.wait_output()
 
-            process = self.container.exec(make_temp_dir_commands)
+            process = self.container.exec(make_temp_dir_command)
             tmp_dir, _ = process.wait_output()
         except ExecError as e:
             logger.exception("Failed to execute commands prior to running backup", exc_info=e)
             raise MySQLExecuteBackupCommandsError(e.stderr)
+        except Exception as e:
+            # Catch all other exceptions to prevent the database being stuck in
+            # a bad state due to pre-backup operations
+            logger.exception("Failed to execute commands prior to running backup", exc_info=e)
+            raise MySQLExecuteBackupCommandsError(e)
 
-        # TODO: remove flags --no-version-check and --no-server-version-check
+        # TODO: remove flags --no-server-version-check
         # when MySQL and XtraBackup versions are in sync
         xtrabackup_commands = " ".join(
-            """
+            f"""
 xtrabackup --defaults-file=/etc/mysql
             --defaults-group=mysqld
             --no-version-check
-            --parallel="$NPROC"
-            --user="$MYSQL_USER"
+            --parallel={nproc.strip()}
+            --user="{self.server_config_user}"
             --password="$MYSQL_PASSWORD"
```

Comment:
xtrabackup will read the password from stdin if the password parameter is empty.
```suggestion
            --password
```
it is ok to handle this in a separate ticket/PR

Reply:
addressed in 2f1db018fed3b71ac85361e7b9ca380e0cc7c95a